srml/support/src/dispatch.rs

```diff
@@ -951,6 +1000,22 @@ macro_rules! decl_module {
 			)*
 		}
 
+		// Implement weight calculation function for Call
+		// no-op wrapper to semantically imply `where $weight: WeighableTransaction` with a meaningful error message.
+		fn __calculate_weight<T: $crate::dispatch::WeighableTransaction>(w: T) -> $crate::dispatch::Weight {
+			w.calculate_weight()
+		}
+		impl<$trait_instance: $trait_name $(<I>, $instance: $instantiable)?> $crate::dispatch::WeighableCall
+			for $call_type<$trait_instance $(, $instance)?>
+		{
+			fn weight(&self) -> $crate::dispatch::Weight {
+				match *self {
+					$( $call_type::$fn_name(..) => __calculate_weight($weight), )*
```

Comment:
```suggestion
					$( $call_type::$fn_name(..) => $crate::dispatch::WeighableTransaction::calculate_weight($weight), )*
```

Should do the same and you don't need a free floating function.

Reply:
I can double check but the pretty sure the wrapper will make the error message much better: e.g. `XXX must implement WeighableTransaction trait`. Otherwise, it would be a bit vaguer (imo) like `cannot find function calculate_weight for XXX`. I'd vote for the first; open to further debate.

Reply:
Yeah, but you see that I also cast the type to a `WeightableTransaction`?